filclient.go

```diff
@@ -1510,24 +1544,44 @@ func (fc *FilClient) RetrieveContentFromPeerWithProgressCallback(
 	startTime := time.Now()
 	totalPayment := abi.NewTokenAmount(0)
 
+	rootCid := proposal.PayloadCID
+	var chanid datatransfer.ChannelID
+	var chanidLk sync.Mutex
+
+	retrievalState := rep.RetrievalState{
+		PayloadCid:        rootCid,
+		PieceCid:          proposal.PieceCID,
+		StorageProviderID: peerID,
+		ClientID:          chanid.Initiator,
+	}
+
 	pchRequired := !proposal.PricePerByte.IsZero() || !proposal.UnsealPrice.IsZero()
 	var pchAddr address.Address
 	var pchLane uint64
 	if pchRequired {
 		// Get the payment channel and create a lane for this retrieval
 		pchAddr, err := fc.getPaychWithMinFunds(ctx, minerWallet)
 		if err != nil {
+			retrievalEvent := rep.RetrievalEvent{
```

Comment:
the lack of these two failure cases was leading to a clean-up leak when relying on only the events to tell us what's going on